tests/components/ui/updatedButton.test.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import "@testing-library/jest-dom";
+import { describe, it, expect } from "vitest";
+import { screen, render } from "@testing-library/react";
+import { Button } from "@/components/ui/button";
+
+describe("Button", () => {
+	it("renders a button with a default props", () => {
+		render(<Button>Click Me</Button>);
+
+		const button = screen.getByRole("button", { name: /click me/i });
+		expect(button).toBeInTheDocument();
+		expect(button).toHaveClass("text-white bg-primary-button");
+	});
+
+	it("renders as child when asChild prop is true", () => {
+		render(
+			<Button asChild>
+				<a href="/">Link Button</a>
```

Comment:
Let's define a different component instead of using something like `asChild`.

These polymorphic components can easily mess up with the semantics and styling, plus we're overloading the button component by allowing it to function as a different element (in this case, a link).

Buttons must perform an action with optional data bound to a form. Links only perform navigation to a URL.